SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestTearOff/ReplicatedTearOffActor.cpp

```diff
@@ -8,14 +8,26 @@ AReplicatedTearOffActor::AReplicatedTearOffActor()
 	bNetLoadOnClient = true;
 	SetReplicatingMovement(true);
 	TestInteger = 0;
+	bFirstTick = true;
+	PrimaryActorTick.bCanEverTick = true;
 }
 
 void AReplicatedTearOffActor::BeginPlay()
 {
 	Super::BeginPlay();
 
-	// Note:  calling TearOff inside the constructor will not prevent the Actor from replicating.
-	TearOff();
+	bFirstTick = true;
+}
+
+void AReplicatedTearOffActor::Tick(float DeltaSeconds)
+{
+	if (bFirstTick)
+	{
+		// Note:  calling TearOff inside the constructor will not prevent the Actor from replicating.
+		// Note2: calling TearOff inside BeginPlay will produce an error if using the ReplicationGraph.
```

Comment:
Curious what the error is? And does it produce an error in native replication graph also?

Reply:
Yeah, produces an error with any replication graph. Took some digging to find this.
Basically, when spawning a *dynamic* actor, the begin play is called immediately upon spawn, so we are still in the actor creation flow. It turns out actors are added to replication lists *just* after beginPlay is called on them. So tearing off an actor with replication graph will attempt to remove it from the replication list, but can't find it there, producing an error.

Reply:
Might be worth adding that to this comment - as it's useful context.